components/constellation/timer_scheduler.rs

```diff
@@ -103,6 +42,18 @@ impl PartialEq for ScheduledEvent {
     }
 }
 
+fn recv_with_timeout<T: Send>(port: &Receiver<T>, from: NsDuration, timeout: NsDuration) -> Option<T> {
```

Comment:
Agreed, it was to avoid another precise_time_ns call. I'll see how I can refactor it so it's cleaner :)